### DISCUSSION TITLE
PocketHost 0.7.2 Released

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#2710] [Show and tell]
# 0.7.2

[PocketHost](https://pockethost.io), the zero-config 'PocketBase in the cloud' platform, has reached v0.7.2 today. Thank you for everyone's continued support for this project.

## What's New

With this release, you can now adjust the PocketBase version. It will accept semantic versions and semantic version ranges.

With limited testing, it seems that PocketBase will refuse to run on an older version of the database, so there is no danger to accidentally rolling back to an earlier version of PocketBase. @ganigeorgiev are you able to confirm this is true?

## Change log

- feat: css cosmetics
- feat: update pocketbase semver
- docs: roadmap
- fix: semver validation
- chore: cleanup backup/restore UI
- chore: remove resetBackups RPC
- chore: backup/restore cleanup
- fix: missing WorkerLogFields type
- chore: log context update
- chore: pm2 log update
- chore: remove instance backup remanence
- chore: singleton base config
- chore: add log breadcrumb support

## Stats and Info

- 326 github stars (personal best!)
- ~2000 registered users
- ~2500 instances created
- ~1.5 million instance invocations to date (# times PocketBase executable has been launched)
- ~400 instances active and used in the past week. This tracks, about 15% of instances are in active use.

Check out the [FAQ](https://pockethost.gitbook.io/manual/overview/faq) and [roadmap](https://pockethost.gitbook.io/manual/overview/roadmap) for more details.

**Comment by ganigeorgiev** [comment]
> With limited testing, it seems that PocketBase will refuse to run on an older version of the database, so there is no danger to accidentally rolling back to an earlier version of PocketBase. @ganigeorgiev are you able to confirm this is true?

There is no builtin mechanism in PocketBase for rolling back versions. We don't have database versioning because it can change by external migrations or even during runtime (_eg. when making collection changes from the Admin UI_).

**Reply by benallfree** [reply]
Sorry, I wasn't clear. How does PocketBase detect whether the current system/internal database tables are compatible? My understanding is that it will automigrate now, so then if a previous binary version is run against an updated database, what will happen? Is the behavior defined?

**Reply by ganigeorgiev** [reply]
> How does PocketBase detect whether the current system/internal database tables are compatible? 

The database schema is not tied to the PocketBase binary version. The only thing that PocketBase do when you start the executable is to check whether all current system and user migrations are applied (_the information for this is stored in the `_migrations` table_). You can manually apply migrations using `migrate up` and revert them in the order they were applied with `migrate down [n]`.

>  if a previous binary version is run against an updated database, what will happen? Is the behavior defined?

We don't apply automatic rollback as it can result in data loss. The behavior is not defined as it depends what was the previously applied version. Most of the time it will just work, but if for example a newer version had deleted or renamed a system column and the old version expects it, then it will fail with runtime error and eventually panic causing the application to restart.
